modules/tribes/client/components/JoinButton.component.js

```diff
@@ -0,0 +1,108 @@
+  const { t } = useTranslation('tribes');
+
+  const ariaLabel = (isMember) ? t('Leave Tribe') : t(`${joinLabel} ({{label}})`, { label: tribe.label });
+  const title = (isMember) ? t(joinedLabel) : t(joinLabel);
+
+  if (!isLogged) {
+    return <a
+      href={`/signup?tribe=${tribe.slug}`}
+      type="button"
+      className="btn btn-sm btn-default tribe-join"
+    >
+      <i className="icon-plus" /> {title}
+    </a>;
+  }
+
+  const leaveTooltip = <Tooltip id={`tribe-${tribe._id}`} placement="bottom">{t('Leave Tribe')}</Tooltip>;
+  const btn = <button
+    type="button"
+    className={`${isMember ? 'btn-active' : ''} btn btn-sm btn-default tribe-join`}
+    onClick={onToggle}
+    disabled={isLoading}
+    aria-label={ariaLabel}
+  >
+    <i className={(isMember) ? 'icon-ok' : 'icon-plus'} /> {title}
+  </button>;
+
+  if (isMember) {
+    return <OverlayTrigger placement="bottom" overlay={leaveTooltip}>{btn}</OverlayTrigger>;
+  } else {
+    return btn;
+  }
+}
+
+JoinButtonPresentational.propTypes = {
+  isMember: PropTypes.bool.isRequired,
+  isLoading: PropTypes.bool,
+  isLogged: PropTypes.bool.isRequired,
+  joinLabel: PropTypes.string,
+  joinedLabel: PropTypes.string,
+  tribe: PropTypes.object.isRequired,
+  onToggle: PropTypes.func
+};
+
+// @TODO this can (and should) be replaced by other container, when we finish the migration; when we start using redux etc.
+export default function JoinButton({ tribe, user, onUpdated, ...rest }) {
+  // isLeaving controls whether the tribe delete modal is shown
+  const [isLeaving, setIsLeaving] = useState(false);
+
+  const [isUpdating, setIsUpdating] = useState(false);
+
+  const isSelfMember = user && user.memberIds && user.memberIds.indexOf(tribe._id) > -1;
+  const [isMember, setIsMember] = useState(isSelfMember);
+
+  async function handleToggleMembership() {
+    if (isUpdating) {
+      return;
+    }
+
+    if (isMember) {
+      setIsLeaving(true);
+    } else {
+      // updating starts
+      setIsUpdating(true);
+
+      // join
+      const data = await api.join(tribe._id);
+      // update the membership locally
+      setIsMember(true);
+
+      // updating finished
+      setIsUpdating(false);
+      onUpdated(data);
+    }
+  }
+
+  async function handleLeave() {
+    setIsUpdating(true);
+    const data = await api.leave(tribe._id);
+    setIsUpdating(false);
+    setIsLeaving(false);
+    setIsMember(false);
+    onUpdated(data);
+  }
+
+  function handleCancelLeave() {
+    setIsLeaving(false);
+  }
+
+  return <>
+    <LeaveTribeModal show={isLeaving} tribe={tribe} onConfirm={handleLeave} onCancel={handleCancelLeave}/>
+    <JoinButtonPresentational tribe={tribe} isLogged={!!user} isMember={isMember} isLoading={isUpdating} {...rest} onToggle={handleToggleMembership} />
+  </>;
+}
+
+JoinButton.propTypes = {
+  tribe: PropTypes.object.isRequired,
+  user: PropTypes.object.isRequired,
```

Comment:
I think user is not required (in the case of logged out user).

Reply:
That's true. Was causing warnings when logged out. Fixed in c5c33f984a7cdb75721ceed50cd222bc2b55b327.